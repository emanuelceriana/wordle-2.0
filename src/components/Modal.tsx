import { ReactNode } from "react";
import "./Modal.css";

interface ModalProps {
  isOpen: boolean;
  onCloseRequest?: () => void;
  showCloseButton: boolean;
  children: ReactNode | ReactNode[];
}

const Modal = ({
  isOpen,
  onCloseRequest,
  showCloseButton,
  children,
}: ModalProps) => {
  if (!isOpen) {
    return null;
  }

  return (
    <>
      <div className="modalBackground" />
      <div className="modal">
        {showCloseButton && (
          <button className="closeButton" onClick={onCloseRequest}>
            X
          </button>
        )}
        <div className="body">{children}</div>
      </div>
    </>
  );
};

export default Modal;
