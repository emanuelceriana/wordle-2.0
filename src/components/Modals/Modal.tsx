import { ReactNode } from "react";
import styles from "./Modal.module.scss";

interface ModalProps {
  isOpen: boolean;
  onCloseRequest?: () => void;
  showCloseButton: boolean;
  title?: string;
  children: ReactNode | ReactNode[];
}

export const Modal = ({
  isOpen,
  onCloseRequest,
  showCloseButton,
  title,
  children,
}: ModalProps) => {
  if (!isOpen) {
    return null;
  }

  return (
    <>
      <div className={styles.modalBg} />

      <div className={styles["modalBg__modal"]}>
        <div className={styles["modalBg__modal-title"]}>{title}</div>
        {showCloseButton && (
          <button
            className={styles["modalBg__modal-cButton"]}
            onClick={onCloseRequest}
          >
            X
          </button>
        )}
        <div className={styles["modalBg__modal-body"]}>{children}</div>
      </div>
    </>
  );
};

export default Modal;
