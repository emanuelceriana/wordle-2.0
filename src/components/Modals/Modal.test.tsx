import { render } from "@testing-library/react";
import { Modal } from "./Modal";

interface ModalProps {
  isOpen: boolean;
  onCloseRequest: () => void;
  showCloseButton: boolean;
  title: string;
}

describe("<MainMenu />", () => {
  let modalProps: ModalProps = {
    isOpen: true,
    onCloseRequest: () => {},
    showCloseButton: true,
    title: "Modal",
  };

  it("should render without errors", () => {
    const { asFragment } = render(<Modal {...modalProps}>test</Modal>);
    expect(asFragment()).toMatchSnapshot();
  });

  it("should be closed if mainMenu is false", () => {
    modalProps.isOpen = false;

    const { queryByText } = render(<Modal {...modalProps}>test</Modal>);

    const modalContent = queryByText("test");

    expect(modalContent).not.toBeInTheDocument();
  });
  it("should be open if helMenuOpen is true", () => {
    modalProps.isOpen = true;

    const { queryByText } = render(<Modal {...modalProps}>test</Modal>);

    const modalContent = queryByText("test");

    expect(modalContent).toBeInTheDocument();
  });
  it("should show close button when showCloseButton is true", () => {
    modalProps.showCloseButton = true;

    const { getByRole } = render(<Modal {...modalProps}>test</Modal>);

    const closeButton = getByRole("button");

    expect(closeButton).toBeDefined();
  });
  it("should not show close button when showCloseButton is false", () => {
    modalProps.showCloseButton = false;

    const { queryByRole } = render(<Modal {...modalProps}>test</Modal>);

    const closeButton = queryByRole("button");

    expect(closeButton).toBe(null);
  });
});
