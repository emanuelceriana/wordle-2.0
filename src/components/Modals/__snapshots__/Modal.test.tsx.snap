// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainMenu /> should render without errors 1`] = `
<DocumentFragment>
  <div
    class="modalBg"
  />
  <div
    class="modalBg__modal"
  >
    <div
      class="modalBg__modal-title"
    >
      Modal
    </div>
    <button
      class="modalBg__modal-cButton"
    >
      X
    </button>
    <div
      class="modalBg__modal-body"
    >
      test
    </div>
  </div>
</DocumentFragment>
`;
